### DISCUSSION TITLE
Where is "local environment" data stored ?

### DISCUSSION THREAD

**Discussion by palansher** [thunderclient/thunder-client-support#636] [Q&A] [answered]
Hello!

I plan to try to use "Local Environment" to store passwords.
I read README and still have no idea: where this sensitive data will be stored:
- on my local Windows PC where VS Code installed
-  on remote ssh Linux machine. I use vscode remote-ssh
- which directory
- stored encrypted or clear text

?
I need to know it for backup and security purposes.

Thank you guys  for the useful app!

**Comment by rangav** [comment, marked as answer]
@palansher the local environment data is stored locally on your pc.

Location path
`%APPDATA%\Code\User\globalStorage\rangav.vscode-thunder-client`

See on remote pc first, if not check in local pc in the above location

**Reply by palansher** [reply]
Thank you!
It really helped.

Some details:
In my env I found empty file on Windows10: 
`c:\Users\vp\AppData\Roaming\Code\User\globalStorage\rangav.vscode-thunder-client\thunderLocalEnvironment.db`

And found file with real data on my linux box: 
`/home/vladp/.vscode-server/data/User/globalStorage/rangav.vscode-thunder-client/thunderLocalEnvironment.json`
It is because I use Windows 10 vscode + remote ssh workplace on Linux VPS.


But the path /home/vladp/.vscode-server/data/User/globalStorage seems not suitable and reliable for backup, considering probability of vs code reinstallation.

It is good to have the possibility to keep local environment data in a custom, more persistent  folder.
For example: /home/user/.rangav.vscode-thunder-client/thunderLocalEnvironment.db

**Reply by rangav** [reply]
thanks for feedback 

Re-installing vscode will not effect the files in global storage.

**Reply by palansher** [reply]
> Re-installing vscode will not effect the files in global storage.

I created  (and saved, yes!!) test Global storage, but there a no changes on file system in folders mentioned above.
So, where is global env data is stored?
Such info it is good to put such info into README, I think.


![image](https://user-images.githubusercontent.com/24850253/175827561-375bcdf4-da49-4ad8-8e83-5c91ac6a4536.png)

**Reply by rangav** [reply]
It is stored in `thunderEnvironment.json` In the global storage folder

**Reply by rangav** [reply]
You can also save all the requests data to `git` workspace In case if you are not aware. 

For more info, see documentation

**Reply by gabrielsantosblanchet** [reply]
Can I have the location on Mac please ?

**Reply by rangav** [reply]
Mac location 
`/Users/{{userName}}/Library/Application Support/Code/User/globalStorage/rangav.vscode-thunder-client/`

replace {{username}} with yours

**Reply by gabrielsantosblanchet** [reply]
thanks
